lib/Dancer2/Core/Request.pm

```diff
@@ -653,6 +653,14 @@ sub _shallow_clone {
     return $new_request;
 }
 
+
+sub route {
+    my ($self, $route) = @_;
+    if ($route) { $self->{route_matched} = $route; }
+
+    return $self->{route_matched};
```

Comment:
Oh yeah. I did that. Sorry.

It was moved to be a class on its own because it needs to inherit from `Plack::Request`. It can't do that with Moo because it will not allow some things. The commits should provide more information on that.

I would at least make a read-only attribute from this. Let me get home and add a different suggestion branch.